src/types.ts

```diff
@@ -10,12 +7,18 @@ export interface InitOptions {
     parseError?: (e: any) => any;
   };
 }
-export interface MonitorOptions<T> {
+
+export interface MonitorOptionsBase {
   context?: Record<string, any>;
-  parseResult?: (r: Unpromisify<T>) => any;
```

Comment:
The reasoning of putting logResult and logExecutionStart only as global was that was to simplify specific method call interface.
In practice you usually want to log all results, and when you try to understand which method did not complete you turn on logExcutionStart also for all methods.

So let's keep logResult and logExecutionStart only as global option

Reply:
So that means we can have:

GlobalOptions {
  logResult
  logExecutionStart
  parseError
}

MethodOptions {
  parseResult
  parseError
}

Without the need of a OptionsBase type

Reply:
I think global options and local options should offer the same capabilities (other than parseResult which is generic). this is a common practice in which you may have global options but local options override them. for example you want to log all results (global) but for one of the functions you don't want to log the result (because it's sensitive or too big or etc)

Reply:
Yeah, I agree with niv, why not give the option to override per function, it's optional